match/tensor.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from logging import info
+
+from .tensordata import TensorData
+
+
+class Tensor(object):
+    def __init__(self, data: TensorData, children: tuple = ()) -> None:
+        """A matrix object that tracks computations for computing gradients."""
+        super().__init__()
+        self.nrow = data.nrow
+        self.ncol = data.ncol
+        self.shape = (self.nrow, self.ncol)
+
+        self.data = data
+        self.grad = List2D(self.nrow, self.ncol, 0.0)
```

Comment:
I'll assume this file has been left untouched since it still reference `List2D`.